diff --git a/belle-sip/src/channel.c b/belle-sip/src/channel.c
index 8a3c088..dcf48c9 100644
--- a/belle-sip/src/channel.c
+++ b/belle-sip/src/channel.c
@@ -582,6 +582,22 @@ void belle_sip_channel_parse_stream(belle_sip_channel_t *obj, int end_of_stream)
 				tmp=*end_of_message;
 				*end_of_message='\0';/*this is in order for the following log to print the message only to its end.*/
 				/*belle_sip_message("channel [%p] read message of [%i] bytes:\n%.40s...",obj, bytes_to_parse, obj->input_stream.read_ptr);*/
+
+// TN hack
+                static const char matchStr[] = "SIP/2.0/TCP";
+                static const char replaceStr[] = "SIP/2.0/TLS";
+                if (bytes_to_parse >= (int)sizeof(matchStr)) {
+                    belle_sip_message("TN hack - attempting TCP to TLS header rewrite");
+                    for (size_t j = 0; j < bytes_to_parse - sizeof(matchStr); ++j) {
+                        if (strncmp(obj->input_stream.read_ptr + j, matchStr, sizeof(matchStr) - 1) == 0) {
+                            memcpy(obj->input_stream.read_ptr + j, replaceStr, sizeof(replaceStr) - 1);
+                            belle_sip_message("TN hack - successful TCP to TLS header rewrite");
+                            break;
+                        }
+                    }
+                }
+// TN hack
+
 				obj->input_stream.msg=belle_sip_message_parse_raw(obj->input_stream.read_ptr
 										,bytes_to_parse
 										,&read_size);
